Source/Data/bin/cook.py

```diff
@@ -19,73 +20,68 @@
 bpmf_hetero = {}
 
 
-if __name__ == '__main__':
-    """bin/cook.py PhraseFreq.txt BPMFMappings.txt BPMFBase.txt data.txt"""
-    if len(sys.argv) < 7:
-        sys.exit('Usage: cook.py phrase-freqs bpmf-mappings bpmf-base punctuations symbols output')
-    #  Read a list of heterophonic singulars and its estimated frequency
-    #  not active yet
-    # try:
-    #     handle = open('heterophony.list', "r")
-    # except IOError as e:
-    #     print("({})".format(e))
-    # while True:
-    #     line = handle.readline()
-    #     if not line: break
-    #     if line[0] == '#': continue
-    #     elements = line.rstrip().split()
-    #     myword = elements[0]
-    #     myvalue = {elements[1], elements[2]}
-    #     if myword in bpmf_hetero:
-    #         bpmf_hetero[myword].append(myvalue)
-    #     else:
-    #         bpmf_hetero[myword] = []
-    #         bpmf_hetero[myword].append(myvalue)
-    # handle.close()
-    #  Reading-in a list of heterophonic words and
-    #  its most frequent pronunciation
+def cook(
+    heterophony1_path,
+    heterophony2_path,
+    heterophony3_path,
+    phrase_freq_path,
+    bpmf_mappings_path,
+    bpmf_base_path,
+    punctuations_path,
+    symbols_path,
+    macros_path,
+    output_path,
+):
     try:
-        handle = open('heterophony1.list', "r")
+        handle = open(heterophony1_path, "r")
     except IOError as e:
         print("({})".format(e))
     while True:
         line = handle.readline()
-        if not line: break
-        if line[0] == '#': continue
+        if not line:
+            break
+        if line[0] == "#":
+            continue
         elements = line.rstrip().split()
         # if elements[0] in bpmf_hetero: break
         bpmf_phon1[elements[0]] = elements[1]
     handle.close()
     try:
-        handle = open('heterophony2.list', "r")
+        handle = open(heterophony2_path, "r")
     except IOError as e:
         print("({})".format(e))
     while True:
```

Comment:
suggestion: I see this file-reading loop repeats a few times, we can extract to a method.